source/org/zfin/marker/presentation/MarkerChromosomalLocationController.java

```diff
@@ -0,0 +1,94 @@
+package org.zfin.marker.presentation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.*;
+import org.zfin.framework.presentation.InvalidWebRequestException;
+import org.zfin.mapping.GenomeLocation;
+import org.zfin.marker.repository.MarkerRepository;
+import org.zfin.profile.presentation.ChromosomalLocationBean;
+import org.zfin.profile.presentation.ChromosomalLocationBeanValidator;
+import org.zfin.profile.repository.ProfileRepository;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+@RequestMapping("/marker")
+public class MarkerChromosomalLocationController {
```

Comment:
I would use 
@RestController
for this class and then use JsonView to configure the response JSON.